libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -316,16 +313,18 @@ uint8_t Compass::handle_mag_cal_command(const mavlink_command_long_t &packet)
         uint8_t mag_mask = packet.param1;
         
         if (mag_mask == 0) { // 0 means all
-            if(!accept_calibration_all()) {
+            if(!_accept_calibration_mask(0xFF)) {
                 result = MAV_RESULT_FAILED;
             }
-            break;
         }
         
-        if(!accept_calibration_mask(mag_mask)) {
+        if(!_accept_calibration_mask(mag_mask)) {
```

Comment:
This will run even when *mag_mask* is 0. Maybe change the previous code to:

```
if (mag_mask == 0) {
  mag_mask = 0xFF;
}
```

and let this one stay as is.